### PR TITLE
unix: delay fs req register until after validation

### PR DESCRIPTION
**Pull request by cjihrig**
On Unix, if a fs function fails validation after `INIT` but before sending the work to the thread pool, then is is necessary to manually unregister the request. This commit moves the registration to just before the work submission. This also makes Unix match the Windows behavior.

Refs: https://github.com/libuv/libuv/pull/1747

### CODE REVIEW

File: src/unix/fs.c [hunk 1]
```diff
@@ -88,10 +86,8 @@
       req->path = path;                                                       \
     } else {                                                                  \
       req->path = uv__strdup(path);                                           \
-      if (req->path == NULL) {                                                \
-        uv__req_unregister(loop, req);                                        \
+      if (req->path == NULL)                                                  \
         return UV_ENOMEM;                                                     \
```
Comment (bnoordhuis) [inline src/unix/fs.c]: Now that I look at it... I think there is an accidental `req->bufs`  memory leak here.  We free it in `uv__fs_work()` but that doesn't run on error and `uv_fs_req_cleanup()` doesn't free it at all.

Suggestion: set `req->bufs = NULL` in INIT and add a `if (req->bufs != req->bufsml) uv__free(req->bufs)` stanza in `uv_fs_req_cleanup()`.

Existing bug so fine to do in a separate commit or PR.

**Comment by cjihrig** [comment]
CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/718/

File: src/unix/fs.c (hunk 1 above)
Comment (cjihrig) [reply src/unix/fs.c]: I'll send a follow up PR.
